### PR TITLE
8307134: Add GTS root CAs

### PR DESCRIPTION
**Pull request by jianglizhou**
This PR was requested by awarner@google.com. The updates were provided by awarner@google.com.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307134](https://bugs.openjdk.org/browse/JDK-8307134): Add GTS root CAs

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**)

### Contributors
 * Andy Warner `<awarner@google.com>`
 * Rajan Halade `<rhalade@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13754/head:pull/13754` \
`$ git checkout pull/13754`

Update a local copy of the PR: \
`$ git checkout pull/13754` \
`$ git pull https://git.openjdk.org/jdk.git pull/13754/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13754`

View PR using the GUI difftool: \
`$ git pr show -t 13754`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13754.diff">https://git.openjdk.org/jdk/pull/13754.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13754#issuecomment-1531801012)

### CODE REVIEW

**Comment by seanjmullan** [comment]
Ideally, an infra test for testing test certs should also be added. @rhalade may be able to contribute this.

**Comment by rhalade** [comment]
> 

I have those implemented. @jianglizhou, please check https://github.com/openjdk/jdk/compare/master...rhalade:jdk:googletrust-certify?expand=1

File: src/java.base/share/data/cacerts/globalsigneccrootcar4
Comment (rhalade) [inline src/java.base/share/data/cacerts/globalsigneccrootcar4:3]: Why is this certificate changed?

**Comment by jianglizhou** [comment]
> Ideally, an infra test for testing test certs should also be added. @rhalade may be able to contribute this.

Thanks!

**Comment by jianglizhou** [comment]
> > 
> 
> I have infra tests for interop implemented. @jianglizhou, please check https://github.com/openjdk/jdk/compare/master...rhalade:jdk:googletrust-certify?expand=1

@rhalade, thanks! I have a minor comment below for your test/jdk/security/infra/java/security/cert/CertPathValidator/certification/GoogleCA.java test.  I'll defer to @awarner@google.com for detailed review, as I don't have much context.

Please fix the bug id, `8303394`:
```
/*
 * @test
 * @bug 8303394
 * @summary Interoperability tests with Google's GlobalSign R4 and GTS Root certificates
 ...
```

Could you please also let me know your plan on committing the GoogleCA.java? Do you plan to create a PR?

**Comment by rhalade** [comment]
> > > 
> > 
> > 
> > I have infra tests for interop implemented. @jianglizhou, please check https://github.com/openjdk/jdk/compare/master...rhalade:jdk:googletrust-certify?expand=1
> 
> @rhalade, thanks! I have a minor comment below for your test/jdk/security/infra/java/security/cert/CertPathValidator/certification/GoogleCA.java test. I'll defer to @[awarner@google.com](mailto:awarner@google.com) for detailed review, as I don't have much context.
> 
> Please fix the bug id, `8303394`:
> 
> ```
> /*
>  * @test
>  * @bug 8303394
>  * @summary Interoperability tests with Google's GlobalSign R4 and GTS Root certificates
>  ...
> ```
> 
> Could you please also let me know your plan on committing the GoogleCA.java? Do you plan to create a PR?

You can include this contribution in your PR. Then it will be easier to backport to JDK 20u as one changeset. I updated bug id in the changeset.

**Comment by aww-aww** [comment]
> > 
> 
> I have infra tests for interop implemented. @jianglizhou, please check https://github.com/openjdk/jdk/compare/master...rhalade:jdk:googletrust-certify?expand=1

Aside from the bug number @jianglizhou raised, the interop tests look good to me.

File: src/java.base/share/data/cacerts/globalsigneccrootcar4
Comment (seanjmullan) [reply src/java.base/share/data/cacerts/globalsigneccrootcar4:3]: Thanks for the explanation. Please file a different issue for this change, since it is outside the scope of this issue, which is to specifically add new roots that have been approved by the Java SE CA Root Program processes. Updated roots, even for small changes such as this, should be handled and approved using an equivalent process.

**Comment by jianglizhou** [comment]
> You can include this contribution in your PR. Then it will be easier to backport to JDK 20u as one changeset. I updated bug id in the changeset.

Done. Please double check.

I ran the GoogleCA.java test with a test JDK binary built with this PR changes included. The test passed.

File: src/java.base/share/data/cacerts/globalsigneccrootcar4
Comment (jianglizhou) [reply src/java.base/share/data/cacerts/globalsigneccrootcar4]: Reverted src/java.base/share/data/cacerts/globalsigneccrootcar4 in this PR. Looks like the update for "globalsigneccrootcar4 [jdk]" in test/jdk/sun/security/lib/cacerts/VerifyCACerts.java also needs to be reverted, otherwise the test fails with the following error. I'll go ahead and revert that as well.

```
ERROR: wrong checksum72:03:89:C2:7B:BF:87:87:E1:65:44:6E:43:5C:65:FF:B5:E8:F9:4C:8A:D1:63:6D:D1:91:4C:AD:1C:9A:CB:3B
Expected checksum23:6E:7A:1C:37:AD:82:31:FD:32:E8:31:63:4B:1A:88:BA:1A:4D:F6:D3:91:CD:0F:B4:09:EC:55:9A:B2:01:51
ERROR: globalsigneccrootcar4 [jdk] SHA-256 fingerprint is incorrect
java.lang.RuntimeException: At least one cacert test failed
        at VerifyCACerts.main(VerifyCACerts.java:380)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
        at java.base/java.lang.Thread.run(Thread.java:1592)
```

Comment (jianglizhou) [reply src/java.base/share/data/cacerts/globalsigneccrootcar4]: Done. Also updated 'CHECKSUM' value in test/jdk/sun/security/lib/cacerts/VerifyCACerts.java reflectively.

**Comment by jianglizhou** [comment]
> Looks good. Please also wait for approval from @rhalade before integrating.

Thanks @seanjmullan. Will wait for @rhalade's approval as well.

**Review by rhalade — APPROVED** [review]
Looks good to me. Please don't forget to update the release note and backport to JDK 20u.

**Pull request closed without merging**
